core/src/display_object/edit_text.rs

```diff
@@ -1977,19 +1996,9 @@ impl<'gc> TDisplayObject<'gc> for EditText<'gc> {
                     && visible_selection.start() == 0
                     && Utc::now().timestamp_subsec_millis() / 500 == 0
                 {
-                    let caret = context.transform_stack.transform().matrix
-                        * Matrix::create_box(
-                            1.0,
-                            edit_text
-                                .text_spans
-                                .default_format()
-                                .size
-                                .unwrap_or_default() as f32,
-                            0.0,
-                            Twips::from_pixels(-1.0),
-                            Twips::from_pixels(2.0),
-                        );
-                    context.commands.draw_rect(Color::BLACK, caret);
+                    let format = edit_text.text_spans.default_format();
+                    let caret_height = format.size.map(Twips::from_pixels).unwrap_or_default();
+                    self.render_caret(context, Twips::ZERO, caret_height, Color::BLACK);
```

Comment:
This code is not perfect... It is executed when the editable text is being rendered without any layout_box in order to show the caret. But without any layout box we do not have information related to the font, etc. and `caret_height` is only an estimate which is often wrong.

From my testing this is the case when an empty editable text is spawned. But when any text is added and removed, now we do have an empty layout box, and the caret is rendered correctly. In my opinion it would be wise to remove this code altogether and replace it with the case of one empty layout box (forcing EditText to always have at least one layout box), but I'm not sure whether it's right and how to do that, and so I have left it as is.

Reply:
>  In my opinion it would be wise to remove this code altogether and replace it with the case of one empty layout box (forcing EditText to always have at least one layout box)

Yeah that sounds just about what I also would have suggested.

> but I'm not sure whether it's right and how to do that, and so I have left it as is.

It's okay, we can do that later as well.